source/Halibut/ServiceModel/RequestCancellationTokens.cs

```diff
@@ -11,28 +11,37 @@ public RequestCancellationTokens(CancellationToken connectingCancellationToken,
         {
             ConnectingCancellationToken = connectingCancellationToken;
             InProgressRequestCancellationToken = inProgressRequestCancellationToken;
-        }
-
-        public CancellationToken ConnectingCancellationToken { get; set; }
-        public CancellationToken InProgressRequestCancellationToken { get; set; }
 
-        public CancellationToken LinkedCancellationToken => LazyLinkedCancellationToken.Value;
-
-        Lazy<CancellationToken> LazyLinkedCancellationToken => new (() =>
-        {
             if (ConnectingCancellationToken == CancellationToken.None && InProgressRequestCancellationToken == CancellationToken.None)
             {
-                return CancellationToken.None;
+                LinkedCancellationToken = CancellationToken.None;
+            }
+            else if (InProgressRequestCancellationToken == CancellationToken.None)
```

Comment:
Optimisations to not wrap a single cancellation token in a new cancellation token source